packages/features/ee/dsync/lib/handleGroupEvents.ts

```diff
@@ -0,0 +1,102 @@
+import type { DirectorySyncEvent, Group } from "@boxyhq/saml-jackson";
+
+import jackson from "@calcom/features/ee/sso/lib/jackson";
+import { getTranslation } from "@calcom/lib/server/i18n";
+import prisma from "@calcom/prisma";
+import { MembershipRole } from "@calcom/prisma/enums";
+import { getTeamOrThrow } from "@calcom/trpc/server/routers/viewer/teams/inviteMember/utils";
+import type { UserWithMembership } from "@calcom/trpc/server/routers/viewer/teams/inviteMember/utils";
+
+import createUserAndInviteToOrg from "./users/createUserAndInviteToOrg";
+import inviteExistingUserToOrg from "./users/inviteExistingUserToOrg";
+
+const handleGroupEvents = async (event: DirectorySyncEvent, orgId: number) => {
+  const { dsyncController } = await jackson();
+  // Find the group name associated with the event
+  const eventData = event.data as Group;
+
+  const groupNames = await prisma.dSyncTeamGroupMapping.findMany({
+    where: {
+      directoryId: event.directory_id,
+      groupName: eventData.name,
+      orgId,
+    },
+    select: {
+      teamId: true,
+      groupName: true,
+    },
+  });
+
+  const org = await getTeamOrThrow(orgId, true);
+
+  if (!org) {
+    throw new Error("Org not found");
+  }
+
+  const userIdsToCreateMemberships: number[] = [];
```

Comment:
Consider extracting into a separate function `createOrInviteUsers` - returning a list of userIds with created/invited users for adding memberships on later. Why doesn't the invite flow include adding the membership is another good question.

Reply:
`handleGroupEvents` is mapping users from a directory group to a team within an org. I wanted to keep adding the user to the org and adding the user to a team seperate.

I'll also create a seperate function to get the `userIds` to create memberships for the team.